src/main/java/me/zeroeightsix/kami/mixin/client/render/MixinTileEntitySignRenderer.java

```diff
@@ -21,9 +21,8 @@
     @Redirect(method = "render", at = @At(value = "FIELD", target = "Lnet/minecraft/tileentity/TileEntitySign;signText:[Lnet/minecraft/util/text/ITextComponent;", opcode = Opcodes.GETFIELD))
     public ITextComponent[] getRenderViewEntity(TileEntitySign sign) {
         if (NoRender.INSTANCE.isEnabled() && NoRender.INSTANCE.getSignText().getValue()) {
-            if (mc.currentScreen instanceof GuiEditSign) {
-                if (getTileSign((GuiEditSign) mc.currentScreen).equals(sign))
-                    return sign.signText;
+            if (mc.currentScreen instanceof GuiEditSign && getTileSign((GuiEditSign) mc.currentScreen).equals(sign)) {
```

Comment:
Store `mc.currentScreen` in a variable instead in case if it gets modified in another thread.